### DISCUSSION TITLE
Does MCAP support conversion between .mcap and .las formats?

### DISCUSSION THREAD

**Discussion by hiyyg** [foxglove/mcap#1113] [General] [answered]
Does it support now or will it support in the future?

**Comment by james-rms** [comment, marked as answer]
hi @hiyyg , we're not aware of any existing code to convert code between `.las` and `.mcap`. I don't think it makes sense to build conversion support into the MCAP libraries themselves. I imagine a converter would depend on the MCAP libraries to support reading and writing MCAPs for conversion.

 If you came up with a conversion script or program, you might want to contribute it as example code in the https://github.com/foxglove/mcap repo, for others who have the same problem.

**Reply by jtbandes** [reply]
@hiyyg It looks like there are (at least) two "LAS" file formats. Are you talking about [Log ASCII Standard](https://en.wikipedia.org/wiki/Log_ASCII_standard) or [LASer](https://en.wikipedia.org/wiki/LAS_file_format) files? Either way, James's answer is accurate :)

You can find some examples of how to create MCAP files with point clouds here: https://foxglove.dev/blog/visualizing-point-clouds-with-custom-colors#using-foxglove-schemas

**Reply by hiyyg** [reply]
Thanks for the reply. For `.las`, I mean LASer file format designed for point clouds.
